### ISSUE TITLE
No sound with ALSA on sample rates above 48000 Hz

### ISSUE THREAD

**Issue by e-pirate** [kinetiknz/cubeb#526]
This is an abandoned/unfixed bug repost from Mozilla's Bugzilla. The original report could be found [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1521621) .

If system (sound card) sampling rate changed in ALSA from default value (48000) to something grater, sound stop working in a very odd way. Sampling rate could be changed by adding the following line to /etc/asound.conf (only works with sound cards that support that):

```defaults.pcm.dmix.rate <samling_rate>```

than restart ALSA and FF, open some video.

Firefox 60.8.0esr build from sources on actual Gentoo Linux.

Actual results:

YouTube work / Coub and most of music players no sound: 88200, 176400, 352800, 705600
Coub and most of music players work / YouTube no sound: 96000, 192000, 384000, 768000

At the moment I start playback on the page and no sound produced, the fallowing message can be seen (this is from older version):
[Child 10792, MediaPlayback #2] WARNING: 7fbdd402ac40 OpenCubeb() failed to init cubeb: file /var/tmp/portage/www-client/firefox-60.4.0/work/firefox-60.4.0/dom/media/AudioStream.cpp, line 388
[Child 10792, MediaPlayback #2] WARNING: Decoder=7fbdde253a60 [OnMediaSinkAudioError]: file /var/tmp/portage/www-client/firefox-60.4.0/work/firefox-60.4.0/dom/media/MediaDecoderStateMachine.cpp, line 3750

So, basically, sound work fine on most of YouTube videos and fail at Coub and many other players for the sampling rates that are multiples of 44100 and vice versa for sampling rates that are multiples of 48000. At the same time, mplayer and Audacious work perfect regardless sampling rate.

Google Chrome behaves the same, so I assumed it is related to libvpx (webm) and rebuild FF against system libvpx-1.5.0 - with no luck.

Pauld told me that this is an cubeb issue and some "good fix" needed, but it still present. Currently, more and more recent videos on YT have no sound, while older play fine. Since both coexist, there is no way to choose sampling rate (multiple of 44K2 or 48K) to cover the majority of cases.

**Comment by kinetiknz** [comment]
The ALSA backend is no longer maintained, so someone will need to contribute their time to debug and fix this.  I posted a possible workaround in the linked Bugzilla bug.

If you wanted to try debugging further, I'd suggest starting by building the cubeb repository and experimenting with tools/cubeb-test.  It defaults to 44.1kHz stereo, but you can change that via command line.  Then you'd need to trace the code (in gdb or by adding printfs) to src/cubeb_alsa.c in alsa_stream_init_single and see where it's failing and why.  I'd guess we're bailing out after the snd_pcm_set_params call.

It may be that the simplest solution is to retrofit cubeb_resampler into the ALSA backend.  Or use the PulseAudio backend. 😬

**Comment by dreamcat4** [comment]
> The workaround did a trick.

@e-pirate can you please clarify *which* workaround solved this problem for you here? was it the suggested patch on Bugzilla? or was it switching over to pulse-rust backend?

many thanks

**Comment by e-pirate** [comment]
> > The workaround did a trick.
> 
> @e-pirate can you please clarify _which_ workaround solved this problem for you here? was it the suggested patch on Bugzilla? or was it switching over to pulse-rust backend?
Set this in you FF configuration:
```
